sqlmesh/core/macros.py

```diff
@@ -105,10 +105,14 @@ def __init__(
         self.python_env = python_env or {}
         self._jinja_env: t.Optional[Environment] = jinja_env
         self.macros = {normalize_macro_name(k): v.func for k, v in macro.get_registry().items()}
+
         prepare_env(self.python_env, self.env)
         for k, v in self.python_env.items():
             if v.is_definition:
                 self.macros[normalize_macro_name(k)] = self.env[v.name or k]
+            elif v.is_import and getattr(self.env.get(k), "__wrapped__", None):
```

Comment:
this is true for any function that has a decorator on it, this doesn't seem to be the right way to detect that it's a macro

Reply:
Refactored the way we detect macros @ runtime